### ISSUE TITLE
[3.22.0] [IOS] Can't open keyboard when clicking TextFormField

### ISSUE THREAD

**Issue by chopper985** [flutter/flutter#149805]
### Steps to reproduce

1. `First Screen` 
2. Click `FloatingActionButton`
3. Navigator to `Second Screen` 
4. Swipe Gesture To Back in IOS 
5. Back to `First Screen` 
6. Click `TextFormField`
7. Keyboard doesn't open

### Expected results

- Can open keyboard when clicking `TextFormField`

### Actual results

- Can not open keyboard when clicking `TextFormField`

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const FirstScreen(),
    );
  }
}

class FirstScreen extends StatelessWidget {
  const FirstScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        centerTitle: true,
        title: const Text('First Screen'),
      ),
      body: GestureDetector(
        onTap: () {
          if (FocusScope.of(context).hasFocus) {
            FocusScope.of(context).unfocus();
          }
        },
        child: Container(
          color: Colors.transparent,
          alignment: Alignment.center,
          padding: const EdgeInsets.symmetric(horizontal: 30),
          child: TextFormField(
            decoration: const InputDecoration(hintText: "Text 1"),
          ),
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () {
          Navigator.push(
            context,
            MaterialPageRoute(
              builder: (context) => const SecondScreen(),
            ),
          );
        },
        child: const Icon(Icons.add),
      ), //,
    );
  }
}

class SecondScreen extends StatelessWidget {
  const SecondScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        centerTitle: true,
        title: const Text('Second Screen'),
      ),
      body: const Center(
        child: Text('text 2'),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/flutter/flutter/assets/55568733/b7c4bd73-ee45-491d-8d67-6b2377ed84e3

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
 Flutter (Channel [user-branch], 3.22.0, on macOS 14.1.2 23B92 darwin-arm64,
    locale en-VN)
    ! Flutter version 3.22.0 on channel [user-branch] at
      /Users/tranthaituan/Downloads/flutter
      Currently on an unknown channel. Run `flutter channel` to switch to an
      official channel.
      If that doesn't fix the issue, reinstall Flutter by following instructions
      at https://flutter.dev/docs/get-started/install.
    ! Upstream repository unknown source is not a standard remote.
      Set environment variable "FLUTTER_GIT_URL" to unknown source to dismiss
      this error.
    • Framework revision 5dcb86f68f (4 weeks ago), 2024-05-09 07:39:20 -0500
    • Engine revision f6344b75dc
    • Dart version 3.4.0
    • DevTools version 2.34.3
    • If those were intentional, you can disregard the above warnings; however
      it is recommended to use "git" directly to perform update checks and
      upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/tranthaituan/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • ANDROID_HOME =
      /opt/homebrew/opt/ruby/bin:/opt/homebrew/lib/ruby/gems/3.1.0/bin/:/opt/hom
      ebrew/opt/ruby/bin:/Users/tranthaituan/.rbenv/shims:/Users/tranthaituan/.r
      benv/bin:/opt/homebrew/opt/ruby@2.6/bin:/opt/homebrew/bin:/opt/homebrew/sb
      in:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/s
      bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/
      bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/v
      ar/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinterna
      l/bin:/Library/Apple/usr/bin:/Applications/iTerm
      2.app/Contents/Resources/utilities:/Users/tranthaituan/Downloads/flutter/b
      in:/Users/tranthaituan/.rvm/bin:/Users/tranthaituan/Library/Android/Sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A507
    • CocoaPods version 1.15.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.89.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.90.0

[✓] Connected device (5 available)
    • Chopper IP (mobile)             • 00008120-001A39512210A01E            •
      ios            • iOS 17.5.1 21F90
    • iPhone 15 Pro Max (mobile)      • 064CD791-B274-4883-95B0-053D7BDBE432 •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-17-0 (simulator)
    • macOS (desktop)                 • macos                                •
      darwin-arm64   • macOS 14.1.2 23B92 darwin-arm64
    • Mac Designed for iPad (desktop) • mac-designed-for-ipad                •
      darwin         • macOS 14.1.2 23B92 darwin-arm64
    • Chrome (web)                    • chrome                               •
      web-javascript • Google Chrome 125.0.6422.142
    ! Error: Browsing on the local area network for DaeWouJun. Ensure the device
      is unlocked and attached with a cable or associated with the same local
      area network as this Mac.
      The device must be opted into Developer Mode to connect wirelessly. (code
      -27)

[✓] Network resources
    • All expected network resources are available.
```

</details>

**Comment by jayden320** [comment]
same bug, temporary solution
create a focusNode object, and then replace
```dart
FocusScope.of(context).unfocus()
``` 
with
```dart
_focusNode.unfocus()
```

**Comment by darshankawar** [comment]
Closing this as duplicate of https://github.com/flutter/flutter/issues/148867. 
The fix has been cherry-picked in stable 3.22.2.
